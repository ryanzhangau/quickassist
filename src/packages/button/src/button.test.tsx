import React from 'react'
import { render, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import AnctechButton, { generateClass } from './index'

describe('AnctechButton component.', () => {
  let component: RenderResult, onClick: () => void

  beforeEach(() => {
    onClick = jest.fn()
    component = render(
      <AnctechButton
        label="Button"
        onClick={onClick}
        outline={false}
        data-test-id="anctech-button"
      />
    )
  })

  it('should match <AnctechButton /> snapshot.', () => {
    const { asFragment } = component
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call onClick function when user click button.', async () => {
    const { getByTestId } = component
    const button = getByTestId('anctech-button')
    await userEvent.click(button)

    expect(onClick).toHaveBeenCalled()
  })

  it('should generate "btn btn-primary btn-sm" when color is primary size is small and outline is falsy.', () => {
    expect(generateClass('sm', 'primary', false)).toBe('btn btn-primary btn-sm')
  })

  it('should generate "btn btn-outline-primary btn-sm" when color is primary size is small and outline is true.', () => {
    expect(generateClass('sm', 'primary', true)).toBe('btn btn-outline-primary btn-sm')
  })
})
