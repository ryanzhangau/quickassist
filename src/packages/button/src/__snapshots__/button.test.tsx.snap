// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnctechButton component. should match <AnctechButton /> snapshot. 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary btn-md"
    data-test-id="anctech-button"
  >
    Button
  </button>
</DocumentFragment>
`;
